libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/AdminPrefsManager.java

```diff
@@ -83,27 +78,18 @@ public void setPrefs(JSONObject attribs, UserAccount account) {
 	 * @param account UserAccount instance.
 	 */
 	public void setPrefs(Map<String, String> attribs, UserAccount account) {
-		if (attribs != null) {
-			String sharedPrefPath = ADMIN_PREFS;
-			if (account != null) {
-				sharedPrefPath = ADMIN_PREFS + account.getOrgLevelFilenameSuffix();
-			}
-			final SharedPreferences sp = SalesforceSDKManager.getInstance().getAppContext()
-	                    .getSharedPreferences(sharedPrefPath, Context.MODE_PRIVATE);
-	        final Editor e = sp.edit();
-	        if (attribs != null) {
-	        	final Set<String> keys = attribs.keySet();
-	        	if (keys != null) {
-	        		for (final String key : keys) {
-	    		        e.putString(key, attribs.get(key));
-	        		}
-	        	}
-	        }
-	        e.commit();
-	    }
+        setPrefs(new JSONObject(attribs), account);
 	}
 
-	/**
+    private SharedPreferences getSharedPreferences(UserAccount account) {
+        String sharedPrefPath = ADMIN_PREFS;
+        if (account != null) {
+            sharedPrefPath = ADMIN_PREFS + account.getOrgLevelFilenameSuffix();
```

Comment:
Then we need to use ```account.getUserLevelFilenameSuffix()``` here. And if we're doing this across the board, we'll need a one time upgrade step to migrate an existing org level admin pref file to user level :-(